### PR TITLE
feat: move to pnpm workspaces

### PR DESCRIPTION
**Pull request by travis**
This gets us on the same page as other web3-storage repos. It makes the development workflow more seamless thanks to `workspace:` deps.

It also uses use `publishConfig` to optimize dev workflow: by pointing `main` at the typescript file in development, apps that support autoreloading autoreload when dependencies change - hooray! this gets rid of the `build` step during development and feels very slick.

Additional changes:

1. remove svelte stuff from rollup configs as they broke this build
2. be more careful about importing types with `import type`

In both of these cases this new config resulted in broken builds without these changes.

### CODE REVIEW

**Comment by travis** [comment]
HM not to grinch this but apparently I didn’t test this with the React examples, and create-react-app doesn’t like loading typescript dependencies and doesn’t make it easy to add loaders. We might want to just start using vite for all our examples? 

Our built packages will still work with create-react-app no problem, but it looks like it might not be the right choice for these working examples? 

There is a real tension between the developer convenience of using typescript deps directly in the examples and pedagogical (?) purity of using the built versions of the dependencies to ensure what downstream users see is the same as what we see in development and I'm not entirely sure how I feel about it yet :)

**Comment by travis** [comment]
ok moving the react examples to use vite like solid and vue do fixes the issue above - PR with an example here: https://github.com/web3-storage/w3ui/pull/203

File: examples/react/file-upload/package.json [hunk 1]
```diff
@@ -3,8 +3,8 @@
   "version": "0.0.0",
   "private": true,
   "dependencies": {
-    "@w3ui/react-keyring": "^2.0.0",
-    "@w3ui/react-uploader": "^3.0.0",
+    "@w3ui/react-keyring": "workspace:^2.0.0",
+    "@w3ui/react-uploader": "workspace:^3.0.0",
```
Comment (alanshaw) [inline examples/react/file-upload/package.json:7]: Do we need the version numbers here? We don't use in other web3.storage projects.

**Review by alanshaw — APPROVED** [review]
🚀

File: examples/react/file-upload/package.json (hunk 1 above)
Comment (travis) [reply examples/react/file-upload/package.json:13]: oo good call, looks like we don't - I'll remove those

File: packages/keyring-core/src/index.ts [hunk 2]
```diff
@@ -127,7 +127,7 @@ export interface CreateAgentOptions extends ServiceConfig {}
  * IndexedDB as unextractable `CryptoKey`s.
  */
 export async function createAgent (options: CreateAgentOptions = {}): Promise<Agent> {
-  const dbName = `${DB_NAME}${(options.servicePrincipal != null) ? '@' + options.servicePrincipal.did() : ''}`
+  const dbName = `${DB_NAME}${(options.servicePrincipal != null) ? '@' + options.servicePrincipal.did().toString() : ''}`
```
[the remark this answers is not in the document]
Comment (travis) [reply packages/keyring-core/src/index.ts:130]: it does, but at some point during the development process the linter or tsc was complaining bitterly about not having this here. I just tried removing it and nobody seems to be complaining, so I'll try removing this - if the PR checks below are still happy we can definitely ship without this!

File: packages/vue-keyring/src/providers/Keyring.ts [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
+  registerSpace: InjectionKey<KeyringContextActions['registerSpace']>
+  cancelRegisterSpace: InjectionKey<KeyringContextActions['cancelRegisterSpace']>
+  getProofs: InjectionKey<KeyringContextActions['getProofs']>
+}
+
 /**
  * Injection keys for keyring provider context.
  */
-export const KeyringProviderInjectionKey = {
+export const KeyringProviderInjectionKey: KeyringProviderInjectionKeyType = {
```
[the remark this answers is not in the document]
Comment (travis) [reply packages/vue-keyring/src/providers/Keyring.ts:27]: we don't! I thought this was also needed to placate the linter but it seems happy now, thanks!

**Pull request merged**